run/data/data.go

```diff
@@ -48,32 +48,72 @@ type Schema struct {
 
 // Table is the data about a DB Table.
 type Table struct {
-	Name          string             // the converted name of the table
-	DBName        string             // the original name of the table in the DB
-	Schema        *Schema            `yaml:"-" json:"-"` // the schema this table is in
-	Columns       Columns            // Database columns
-	ColumnsByName map[string]*Column `yaml:"-" json:"-"` // dbname to column
-	PrimaryKeys   Columns            // Primary Key Columns
+	Name                               string                      // the converted name of the table
+	DBName                             string                      // the original name of the table in the DB
+	Schema                             *Schema                     `yaml:"-" json:"-"` // the schema this table is in
+	Columns                            Columns                     // Database columns
+	ColumnsByName                      map[string]*Column          `yaml:"-" json:"-"` // dbname to column
+	ForeignColumnsByForeignKey         map[string][]*ForeignColumn // foreign key columns
+	PrimaryKeys                        Columns                     // Primary Key Columns
+	ForeignKeys                        []string                    // database names of this tables foreign keys
+	ForeignKeyReferences               []string                    // database names of foreign keys referencing this table
+	ForeignTablesByForeignKey          map[string]*ForeignTable    // tables referenced by foreign keys
+	ForeignTablesByForeignKeyReference map[string]*ForeignTable    // all tables referencing this table
```

Comment:
I'm not sure we need ForeignTables ... can you talk about why you think we need this?  If we don't have a specific reason for it, let's leave it out. We can always put it in later if requested, but we can't ever take it out.